File: client/src/types/suggestion.ts
export type SuggestionStatus = "Pending" | "Accepted";

export interface Suggestion {
  id: string;
  bookTitle: string;
  author: string;
  genre: string;
  coverImageUrl?: string;
  fromUserEmail: string;
  toUserEmail: string;
  status: SuggestionStatus;
  createdAt: string;
}

export interface SuggestionDTO {
  id: string;
  bookTitle: string;
  author: string;
  genre: string;
  coverImageUrl?: string;
  fromUserEmail: string;
  status: SuggestionStatus;
  createdAt: string;
}

export interface SuggestionCreateDTO {
  bookId: string;
  toUserEmail: string;
}

File: server/BookTracker.Api/Models/Suggestion.cs
using System.ComponentModel.DataAnnotations;

public enum SuggestionStatus {
    Pending,
    Accepted
}

public class Suggestion {
    public Guid Id { get; set; }

    [Required]
    public string BookTitle { get; set; } = null!;

    [Required]
    public string Author { get; set; } = null!;

    [Required]
    public string Genre { get; set; } = null!;

    public string? CoverImageUrl { get; set; }

    [Required]
    public string FromUserEmail { get; set; } = null!;

    [Required]
    public string ToUserEmail { get; set; } = null!;

    public string? TempPasswordHash { get; set; }

    public SuggestionStatus Status { get; set; } = SuggestionStatus.Pending;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
